src/app-layout/visual-refresh-toolbar/index.tsx

```diff
@@ -1,563 +1,91 @@
-      toolsOpen,
-      tools,
-      toolsWidth,
-      onToolsToggle,
-    });
-
-    const onActiveDrawerChangeHandler = (
-      drawerId: string | null,
-      params: OnChangeParams = { initiatedByUserAction: true }
-    ) => {
-      onActiveDrawerChange(drawerId, params);
-      drawersFocusControl.setFocus();
-    };
-
-    const [splitPanelOpen = false, setSplitPanelOpen] = useControllable(
-      controlledSplitPanelOpen,
-      onSplitPanelToggle,
-      false,
-      {
-        componentName: 'AppLayout',
-        controlledProp: 'splitPanelOpen',
-        changeHandler: 'onSplitPanelToggle',
-      }
-    );
-
-    const onSplitPanelToggleHandler = () => {
-      setSplitPanelAnimationDisabled(false);
-      setSplitPanelOpen(!splitPanelOpen);
-      splitPanelFocusControl.setLastInteraction({ type: splitPanelOpen ? 'close' : 'open' });
-      fireNonCancelableEvent(onSplitPanelToggle, { open: !splitPanelOpen });
-    };
-
-    const [splitPanelPreferences, setSplitPanelPreferences] = useControllable(
-      controlledSplitPanelPreferences,
-      onSplitPanelPreferencesChange,
-      undefined,
-      {
-        componentName: 'AppLayout',
-        controlledProp: 'splitPanelPreferences',
-        changeHandler: 'onSplitPanelPreferencesChange',
-      }
-    );
-
-    const onSplitPanelPreferencesChangeHandler = (detail: AppLayoutProps.SplitPanelPreferences) => {
-      setSplitPanelPreferences(detail);
-      splitPanelFocusControl.setLastInteraction({ type: 'position' });
-      fireNonCancelableEvent(onSplitPanelPreferencesChange, detail);
-    };
-
-    const [splitPanelSize = 0, setSplitPanelSize] = useControllable(
-      controlledSplitPanelSize,
-      onSplitPanelResize,
-      getSplitPanelDefaultSize(splitPanelPreferences?.position ?? 'bottom'),
-      { componentName: 'AppLayout', controlledProp: 'splitPanelSize', changeHandler: 'onSplitPanelResize' }
-    );
-
-    const [splitPanelReportedSize, setSplitPanelReportedSize] = useState(0);
-    const [splitPanelHeaderBlockSize, setSplitPanelHeaderBlockSize] = useState(0);
-
-    const onSplitPanelResizeHandler = (size: number) => {
-      setSplitPanelSize(size);
-      fireNonCancelableEvent(onSplitPanelResize, { size });
-    };
-
-    const [splitPanelToggleConfig, setSplitPanelToggleConfig] = useState<SplitPanelSideToggleProps>({
-      ariaLabel: undefined,
-      displayed: false,
-    });
-
-    const globalDrawersFocusControl = useMultipleFocusControl(true, activeGlobalDrawersIds);
-    const drawersFocusControl = useFocusControl(!!activeDrawer?.id, true, activeDrawer?.id);
-    const navigationFocusControl = useFocusControl(navigationOpen, navigationTriggerHide);
-    const splitPanelFocusControl = useSplitPanelFocusControl([splitPanelPreferences, splitPanelOpen]);
+  const hasToolbar = !!toolbarProps;
+  useEffect(() => {
+    stateManager.current.hasToolbar = hasToolbar;
+    stateManager.current.setToolbar?.(hasToolbar);
+  }, [stateManager, hasToolbar]);
```

Comment:
Important place no. 2 – here we send information back to the async part. Find the other instance of `stateManager.current.setToolbar` so see where it is used